src/main/java/se/sundsvall/billingpreprocessor/service/mapper/ExternalInvoiceMapper.java

```diff
@@ -0,0 +1,220 @@
+
+import se.sundsvall.billingpreprocessor.integration.db.model.AccountInformationEmbeddable;
+import se.sundsvall.billingpreprocessor.integration.db.model.AddressDetailsEmbeddable;
+import se.sundsvall.billingpreprocessor.integration.db.model.BillingRecordEntity;
+import se.sundsvall.billingpreprocessor.integration.db.model.DescriptionEntity;
+import se.sundsvall.billingpreprocessor.integration.db.model.InvoiceEntity;
+import se.sundsvall.billingpreprocessor.integration.db.model.InvoiceRowEntity;
+import se.sundsvall.billingpreprocessor.integration.db.model.RecipientEntity;
+import se.sundsvall.billingpreprocessor.service.creator.definition.external.CustomerRow;
+import se.sundsvall.billingpreprocessor.service.creator.definition.external.FileHeaderRow;
+import se.sundsvall.billingpreprocessor.service.creator.definition.external.InvoiceAccountingRow;
+import se.sundsvall.billingpreprocessor.service.creator.definition.external.InvoiceDescriptionRow;
+import se.sundsvall.billingpreprocessor.service.creator.definition.external.InvoiceFooterRow;
+import se.sundsvall.billingpreprocessor.service.creator.definition.external.InvoiceHeaderRow;
+import se.sundsvall.billingpreprocessor.service.creator.definition.external.InvoiceRow;
+
+public class ExternalInvoiceMapper {
+
+	private ExternalInvoiceMapper() {}
+
+	/**
+	 * method for creating file header row for external invoice files
+	 * 
+	 * @param generatingSystem the generating system for the file
+	 * @param invoiceType      type of invoice
+	 * @return FileHeaderRow for external invoice files
+	 * @throws ThrowableProblem if any mandatory data is missing
+	 */
+	public static FileHeaderRow toFileHeader(String generatingSystem, String invoiceType) {
+		return FileHeaderRow.create()
+			.withGeneratingSystem(ofNullable(generatingSystem).orElseThrow(() -> createProblem(ERROR_GENERATING_SYSTEM_NOT_PRESENT)))
+			.withCreatedDate(LocalDate.now())
+			.withInvoiceType(ofNullable(invoiceType).orElseThrow(() -> createProblem(ERROR_INVOICE_TYPE_NOT_PRESENT)));
+	}
+
+	/**
+	 * Method for mapping recipient data to a customer row for an external invoice file
+	 * 
+	 * @param legalId             legal id of invoice recipient
+	 * @param billingRecordEntity entity representing the billingRecordEntity
+	 * @return CustomerRow for external invoice files representing provided data
+	 * @throws ThrowableProblem if any mandatory data is missing
+	 */
+	public static CustomerRow toCustomer(String legalId, BillingRecordEntity billingRecordEntity) {
+		final var recipientEntity = ofNullable(billingRecordEntity.getRecipient()).orElseThrow(() -> createProblem(ERROR_RECIPIENT_NOT_PRESENT));
+
+		return CustomerRow.create()
+			.withLegalId(ofNullable(legalId).orElseThrow(() -> createProblem(ERROR_LEGALID_NOT_PRESENT)))
+			.withCustomerName(ofNullable(extractRecipientName(recipientEntity)).orElseThrow(() -> createProblem(ERROR_RECIPIENT_NAME_NOT_PRESENT)))
+			.withCareOf(recipientEntity.getAddressDetails().getCareOf())
```

Comment:
This will cause a NPE if the getAddressDetails returns null.

Reply:
The AddressDetails object is an embedded class in the RecipientEntity, which means that if the recipient exists there is bound to be an addressDetail for it. Hence it should not be possible for getAddressDetails to return null here.